### PR TITLE
generator: Looking for description files on install dir and env var

### PR DESCRIPTION
**Pull request by edersondisouza**
v2:
Environment variable using singular;
Env var and install dir exclusive: if using env var, install dir won't be looked at.

Currently, to set description files, one has to use `-j` option. This
patch introduces two changes to this: now sol-fbp-generator looks on
Soletta install dir to find description files. In addition, it's
possible to use env var `SOL_FLOW_DESCRIPTION_PATHS` as well. This last
one can be composed of different colon (:) separated paths to files or
directories.

Signed-off-by: Ederson de Souza <ederson.desouza@intel.com>

### CODE REVIEW

File: src/bin/sol-fbp-generator/main.c [hunk 1]
```diff
@@ -2546,6 +2565,29 @@ main(int argc, char *argv[])
     if (!parse_args(argc, argv))
         goto fail_args;
 
+    /* Let's look for node descriptions environment variable SOL_FLOW_DESCRIPTIONS_PATH */
+    descripions_paths = getenv("SOL_FLOW_DESCRIPTIONS_PATH");
+    if (descripions_paths && descripions_paths[0] != '\n') {
```
Comment (cmarcelo) [inline src/bin/sol-fbp-generator/main.c]: '\n' or 0?

File: src/bin/sol-fbp-generator/main.c [hunk 2]
```diff
@@ -2516,6 +2514,25 @@ write_file(const char *filename, const struct sol_buffer *buf)
...
+{
+    struct sol_str_slice paths = sol_str_slice_from_str(descripions_paths);
+    struct sol_str_slice token;
+    struct sol_str_slice delim = SOL_STR_SLICE_LITERAL(":");
+    const char *itr = NULL;
+
+    while (sol_str_slice_split_iterate(paths, &token, &itr, delim)) {
+        char *path = sol_str_slice_to_str(token);
+        if (!path) {
+            SOL_WRN("Could not parse pathectories from SOL_FLOW_DESCRIPTIONS_PATH environment variable");
```
Comment (cmarcelo) [inline src/bin/sol-fbp-generator/main.c]: Typo "pathectories".

File: src/bin/sol-fbp-generator/main.c (hunk 1 above)
Comment (edersondisouza) [inline src/bin/sol-fbp-generator/main.c]: Ops!

File: src/bin/sol-fbp-generator/main.c (hunk 2 above)
Comment (edersondisouza) [inline src/bin/sol-fbp-generator/main.c]: I must've been drunk... o.O

File: src/bin/sol-fbp-generator/main.c [hunk 3]
```diff
@@ -1888,7 +1888,13 @@ print_usage(const char *program)
...
         "    -s  Define a function named SYMBOL that will return the type from FBP\n"
         "        and don't generate any main function or entry point.\n"
-        "    -I  Define search path for FBP files\n"
+        "    -I  Define search path for FBP files\n\n"
+        "Note:\n"
+        "    Soletta install directory will also be looked for node type description\n"
+        "    files in addition to any `-j` argument, unless environmment variable\n"
+        "    SOL_FLOW_DESCRIPTIONS_PATH is set. In this case, paths on this variable\n"
+        "    will be looked in addition to any `-j` argument.\n"
+        "    SOL_FLOW_DESCRIPTIONS_PATH may contain multiples paths separated by ':'.\n"
```
Comment (cmarcelo) [inline src/bin/sol-fbp-generator/main.c]: "... multiple paths..."

File: src/bin/sol-fbp-generator/main.c [hunk 4]
```diff
@@ -1888,7 +1888,13 @@ print_usage(const char *program)
         "        Multiple -j can be passed.\n"
         "    -s  Define a function named SYMBOL that will return the type from FBP\n"
         "        and don't generate any main function or entry point.\n"
-        "    -I  Define search path for FBP files\n"
+        "    -I  Define search path for FBP files\n\n"
```
Comment (cmarcelo) [inline src/bin/sol-fbp-generator/main.c]: Minor trick: If you put the extra "\n" in its own line, the patch wouldn't touch unrelated text for -I. Not very important though.

File: src/bin/sol-fbp-generator/main.c [hunk 5]
```diff
@@ -2516,6 +2514,25 @@ write_file(const char *filename, const struct sol_buffer *buf)
     return err;
 }
 
+static void
+handle_descriptions_paths(const char *descripions_paths)
```
Comment (cmarcelo) [inline src/bin/sol-fbp-generator/main.c]: Typo: "descripions"

File: src/bin/sol-fbp-generator/main.c [hunk 6]
```diff
@@ -2528,6 +2545,8 @@ main(int argc, char *argv[])
     struct exported_option *exported_opt;
     struct exported_option_description *opt_description;
     struct sol_fbp_node *n;
+    char root_dir[PATH_MAX], descriptions_dir[PATH_MAX];
+    const char *descripions_paths;
```
Comment (cmarcelo) [inline src/bin/sol-fbp-generator/main.c]: Here "descripions"  too.

**Comment by cmarcelo** [comment]
A bunch of small stuff, but patch looks OK to me. Haven't tested locally.

+1 with comments above.

**Comment by edersondisouza** [comment]
Integrated with fixes, tks!

**Pull request closed without merging**